source/common/stats/isolated_store_impl.cc

```diff
@@ -34,21 +33,37 @@ IsolatedStoreImpl::IsolatedStoreImpl(SymbolTable& symbol_table)
         return alloc_.makeTextReadout(name, name, StatNameTagVector{});
       }),
       null_counter_(new NullCounterImpl(symbol_table)),
-      null_gauge_(new NullGaugeImpl(symbol_table)),
-      default_scope_(std::make_shared<ScopePrefixer>("", *this)) {}
+      null_gauge_(new NullGaugeImpl(symbol_table)) {}
+
+ScopeSharedPtr IsolatedStoreImpl::rootScope() {
+  if (lazy_default_scope_ == nullptr) {
+    StatNameManagedStorage name_storage("", symbolTable());
+    lazy_default_scope_ = makeScope(StatName(name_storage.statName()));
```

Comment:
why we need another StatName here when statName() is already returning a StatName?

Reply:
um we don't. removing.